community/logging/src/main/java/org/neo4j/logging/RotatingFileOutputStreamSupplier.java

```diff
@@ -227,10 +236,10 @@ public void run()
         {
             rotationExecutor.execute( runnable );
         }
-        catch ( RejectedExecutionException e )
+        catch ( Exception e )
         {
-            // Run directly instead
-            runnable.run();
+            rotationListener.rotationError( e, outRef.get() );
```

Comment:
Yeah, you are painting a bike shed ;)

JobScheduler is way too broad an interface. This class just needs to be able to execute a runable. Of course, the constructee is free to supply an Executor that is implemented from JobScheduler (which is what happens in practice).

I disagree on your point 2. Any exception is a failure to rotate. The `RotatingFileOutputStreamSupplier#get()` _always_ returns a value: the current stream. Rotation is a lazy side effect it _might_ initiate. And if rotation can't be done, for any reason, the listener is notified and is free to respond in whatever way makes sense in the context this class is used.